diff --git a/node_modules/twitch-webhook/src/index.js b/node_modules/twitch-webhook/src/index.js
index f8938f0..e8e9dcd 100755
--- a/node_modules/twitch-webhook/src/index.js
+++ b/node_modules/twitch-webhook/src/index.js
@@ -40,6 +40,10 @@ class TwitchWebhook extends EventEmitter {
 
     if (!options.callback) {
       throw new errors.FatalError('Callback URL not provided!')
+	}
+
+    if (!options.tokenPromise) {
+      throw new errors.FatalError('Token promise not provided!')
     }
 
     super()
@@ -173,19 +177,19 @@ class TwitchWebhook extends EventEmitter {
         .digest('hex')
 
       requestOptions.qs['hub.secret'] = secret
-    }
-
-    return request
-      .post(requestOptions)
-      .catch(err => {
-        throw new errors.RequestDenied(err)
-      })
-      .then(response => {
-        this._subscriptions[topic] = {}
-        if (this._options.secret) {
-          this._subscriptions[topic].secret = requestOptions.qs['hub.secret']
-        }
-      })
+	}
+
+    return this._options.tokenPromise().then(token => {
+      requestOptions.headers['Authorization'] = `Bearer ${token}`
+      return request.post(requestOptions)
+    }).catch(err => {
+      throw new errors.RequestDenied(err)
+    }).then(response => {
+      this._subscriptions[topic] = {}
+      if (this._options.secret) {
+        this._subscriptions[topic].secret = requestOptions.qs['hub.secret']
+      }
+    })
   }
 
   /**
@@ -374,6 +378,7 @@ class TwitchWebhook extends EventEmitter {
       payload.topic = topic
       payload.options = options
       payload.endpoint = endpoint
+      payload.headers = request.headers;
       payload.event = this._fixDate(topic, data)
 
       this.emit(topic, payload)
